src/client/dfs/SConscript

```diff
@@ -54,19 +56,45 @@ def scons():
                         instver = cache["lustre-client"].installed
             if gotpkg is True:
                 if parse_version(instver.version) >= parse_version('2.12.57'):
```

Comment:
This would be more robustly done as feature tests rather than a version comparison.  Version numbers can be misleading.  The EL7 kernel is an excellent example of this.  It's version number is a **very** unreliable way of knowing that kernel supports.

Reply:
> This would be more robustly done as feature tests

Not sure to understand this. Do you mean it should be tested by trying to compile a small snippet including the necessary Lustre headers and using/calling the specific defines/APIs from the Lustre feature we are looking for ?? Like frequently done with configure/gnu-based builds ??

Reply:
Yes, that is exactly what I mean.  But I don't mean you should write free-form python code to do it but rather use the features built into SCons to do it for you.  SCons is analogous to "configure/gnu-based" (actually called GNU Autoconf or more generally GNU Autotools) builds and provides the same kind of tools for API discovery as Autoconf.

See my next comment for more details.